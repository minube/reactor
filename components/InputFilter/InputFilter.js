import {
  arrayOf, bool, func, shape, string,
} from 'prop-types';
import React, { PureComponent } from 'react';
import { View } from 'react-native';

import Text from '../Text';
import Input from '../Input';
import Touchable from '../Touchable';
import styles from './InputFilter.style';


class InputFilter extends PureComponent {
  static propTypes = {
    color: string,
    dataSource: arrayOf(shape({})),
    disabled: bool,
    hint: string,
    label: string,
    onClickItem: func,
    onChange: func,
    value: string,
  };

  static defaultProps = {
    color: undefined,
    dataSource: [],
    disabled: false,
    hint: undefined,
    label: undefined,
    onClickItem: undefined,
    onChange: undefined,
    value: undefined,
  };

  constructor(props) {
    super(props);
    const { value, dataSource } = props;

    this.searcherTimeout = undefined;
    this.state = { inputValue: value, visible: false, dataFiltered: dataSource };
  }

  _onChange = async (value) => {
    const { dataSource, onClickItem } = this.props;
    this.setState({ inputValue: value, visible: value.length > 2 });

    if (value.length === 0) onClickItem({ id: undefined });

    const dataFiltered = dataSource.filter(item => (
      item.name.toLowerCase().search(value) > -1
    ));

    this.setState({ dataFiltered });
  }

  _onClickItem = (item) => {
    const { onClickItem } = this.props;
    onClickItem(item);
    this.setState({ inputValue: item.name, visible: false });
  };

  _onToggle = (event) => {
    const { state: { visible }, _onToggleOutside } = this;
    const { onClickItem } = this.props;

    if (event) {
      this.touchable = event.currentTarget;
      if (event.target.value && event.target.value.length > 0) onClickItem({ id: undefined })
    }

    if (!visible) document.addEventListener('click', _onToggleOutside, false);
    else document.removeEventListener('click', _onToggleOutside, false);

    this.setState({ inputValue: '', visible: false });
  }

  _onToggleOutside = (event) => {
    const { _onToggle } = this;
    if (this.touchable.contains(event.target)) return;
    _onToggle();
  }

  render() {
    const {
      _onToggle, _onChange, _onClickItem,
      state: {
        inputValue, visible, dataFiltered,
      },
      props: { ...inherit },
    } = this;

    return (
      <View style={[styles.container, inherit.style]} onClick={event => _onToggle(event)}>
        <Input
          {...inherit}
          onChange={values => _onChange(values)}
          value={inputValue}
          style={styles.input}
        />
        {visible && (
          <View style={[styles.list, inherit.label ? styles.marginListLabel : styles.marginList]}>
            {dataFiltered.map((item, index) => (
              <Touchable
                style={[styles.item, index !== dataFiltered.length - 1 && styles.border]}
                key={item.id}
                onPress={() => _onClickItem(item)}
              >
                <Text subtitle level={2} key={item.id} numberOfLines={1}>{item.name}</Text>
              </Touchable>
            ))}
          </View>
        )}
      </View>
    );
  }
}

export default InputFilter;
