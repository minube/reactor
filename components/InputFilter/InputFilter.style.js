import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const {
  BORDER_RADIUS, COLOR, UNIT, SPACE,
} = THEME;

const HEIGHT = UNIT * 4.4;
const HEIGHT_LABEL = UNIT * 6.8;

export default StyleSheet.create({
  input: {
    width: '100%',
  },

  item: {
    paddingVertical: SPACE.MEDIUM,
  },

  border: {
    borderBottomWidth: 1,
    borderColor: COLOR.BASE,
  },

  list: {
    backgroundColor: COLOR.WHITE,
    borderRadius: BORDER_RADIUS,
    maxHeight: UNIT * 50,
    overflow: 'scroll',
    paddingHorizontal: SPACE.MEDIUM,
    position: 'absolute',
    width: '100%',
    zIndex: 3,
    borderColor: COLOR.BASE,
    borderWidth: UNIT * 0.1,
    transform: [{ translateY: -UNIT * 0.1 }],
  },

  marginList: {
    top: HEIGHT,
  },

  marginListLabel: {
    top: HEIGHT_LABEL,
  },
});
