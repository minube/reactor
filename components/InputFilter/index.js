export { default } from './InputFilter'; // eslint-disable-line
