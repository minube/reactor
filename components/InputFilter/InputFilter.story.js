import React, { PureComponent } from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import {
  text, boolean, color, object,
} from '@storybook/addon-knobs/react';
import InputFilter from './InputFilter';

const DATA = [
  {name: 'Aba Travel', id: 1, subtitle: 'Comunidad de Madrid, España'},
  {name: 'Abc4events', id: 2, subtitle: 'Comunidad de Madrid, España'},
  {name: 'Abrera Madrid', id: 3, subtitle: 'Comunidad de Madrid, España'},
  {name: 'Abrera', id: 4, subtitle: 'Comunidad de Madrid, España'},
];

const STYLE = { padding: 10, width: 256 };

_onChange = () => DATA;
_onClickItem = () => {};

class HOC extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { data: props.data };
  }

  _onChange = () => {
    const { onChange } = this.props;
    this.setState({ data: onChange() });
  }

  render() {
    const { _onChange, props, state: { data } } = this;
    return <InputFilter {...props} dataSource={data} onChange={_onChange} />;
  }
}

storiesOf('✅ InputFilter', module)
  .add('default', () => (
    <InputFilter />
  ))
  .add('color', () => (
    <InputFilter color="green" />
  ))
  .add('dataSource', () => (
    <InputFilter dataSource={DATA} />
  ))
  .add('disabled', () => (
    <InputFilter disabled />
  ))
  .add('hint', () => (
    <InputFilter hint="Optional field" />
  ))
  .add('label', () => (
    <InputFilter label="InputFilter" />
  ))
  .add('placeholder', () => (
    <InputFilter placeholder={"Looking for..."} />
  ))
  .add('value', () => (
    <InputFilter value="Madrid" />
  ))
  .add('onChange', () => (
    <HOC onChange={_onChange} data={DATA} />
  ))
  .add('onClickItem', () => (
    <HOC onChange={_onChange} onClickItem={_onClickItem} dataSource={this.data} />
  ))
  .add('🏀 Playground', () => (
    <InputFilter
      color={color('color', undefined)}
      hint={text('hint', 'This is a hint')}
      disabled={boolean('disabled', false)}
      label={text('label', 'This is a label')}
      value={text('value', 'Madrid')}
      onChange={action('InputFilter.onChange()')}
      style={object('style', STYLE)}
    />
  ));
