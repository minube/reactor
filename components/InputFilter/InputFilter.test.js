import React from 'react';
import renderer from 'react-test-renderer';
import InputFilter from './InputFilter';

const VALUE = 'Madrid';
const DATA = [
  {name: 'Madrid', id: 23},
  {name: 'Madrid Snowzone', id: 12},
];
const FUNC = () => {};
const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10 };

describe('<InputFilter>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputFilter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<InputFilter disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {dataSource}', () => {
    const tree = renderer.create(<InputFilter dataSource={DATA} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {hint}', () => {
    const tree = renderer.create(<InputFilter hint="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<InputFilter label="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<InputFilter value={VALUE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onChange}', () => {
    const tree = renderer.create(<InputFilter onChange={FUNC} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onClickItem}', () => {
    const tree = renderer.create(<InputFilter onClickItem={FUNC} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {placeholder}', () => {
    const tree = renderer.create(<InputFilter placeholder={"Hello World"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<InputFilter style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
