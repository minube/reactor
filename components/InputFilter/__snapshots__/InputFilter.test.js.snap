// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFilter> renders 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {dataSource} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={
          Array [
            Object {
              "id": 23,
              "name": "Madrid",
            },
            Object {
              "id": 12,
              "name": "Madrid Snowzone",
            },
          ]
        }
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {disabled} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          Object {
            "backgroundColor": "#f4f4f4",
          },
          false,
          false,
          false,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={false}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            Object {
              "color": "#707070",
            },
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {hint} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontSize": 12,
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "marginTop": 4,
            "maxWidth": "100%",
            "paddingLeft": 0,
          },
        ]
      }
    >
      Hello World
    </Text>
  </View>
</View>
`;

exports[`<InputFilter> when {label} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
            "fontWeight": "400",
            "textAlignVertical": "center",
          },
          Object {
            "fontFamily": undefined,
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 16,
          },
          Object {
            "color": "#707070",
          },
          Object {
            "color": "#4a4a4a",
            "marginBottom": 8,
          },
        ]
      }
    >
      Hello World
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {onChange} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {onClickItem} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onClickItem={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {placeholder} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Hello World"
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {style} 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(0,255,0,0.25)",
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;

exports[`<InputFilter> when {value} 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 20,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#cdd2d4",
            "borderRadius": 2.5,
            "borderWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 8,
          },
          false,
          false,
          false,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit={true}
        dataSource={Array []}
        editable={true}
        keyboard="default"
        keyboardType="default"
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#707070"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
              "color": "#4a4a4a",
              "flex": 1,
              "fontFamily": undefined,
              "fontSize": 16,
              "fontWeight": "400",
              "minHeight": 42,
              "paddingHorizontal": 4,
              "paddingVertical": 7.5,
              "width": "100%",
            },
            false,
            undefined,
            undefined,
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="Madrid"
      />
    </View>
  </View>
</View>
`;
